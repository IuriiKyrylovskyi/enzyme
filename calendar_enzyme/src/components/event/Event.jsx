import React, { useState } from "react";
import { useGlobalContext } from "../../context";
import { MdDelete } from "react-icons/md";
import "./event.scss";

const Event = ({ id, height, marginTop, title, time, handleDelete, fetchEvents }) => {
  const { isEvent, isOpen } = useGlobalContext();

  const [isClicked, setIsClicked] = useState(false);

  const onOpenDelete = (e) => {
    if (isOpen) {
      return;
    }
    // if (e.target.className === 'event') {
    //   return setIsClicked(!isEvent);
    // }
    if (!isEvent && !isClicked) {
      return setIsClicked(!isEvent);
    }
    console.log(height);
    return setIsClicked(!isClicked);
  };

  function onCloseDelete() {
    handleDelete(id);
    fetchEvents();
    setIsClicked(false);
  }

  const eventStyle = {
    height,
    marginTop,
  };

  return (
    <>
      <div
        style={eventStyle}
        className="event"
        onClick={onOpenDelete}
        //
      >
        <div className="event__title">{title}</div>
        <div className="event__time">{time}</div>
        {isClicked && (
          <span
            className="delete-event-btn"
            style={{ top: height - 10 }}
            onClick={onCloseDelete}
            //
          >
            <MdDelete /> Delete
          </span>
        )}
      </div>
    </>
  );
};

export default Event;
