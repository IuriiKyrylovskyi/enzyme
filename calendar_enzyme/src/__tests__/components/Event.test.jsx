import { shallow } from "enzyme";
import React from "react";
import Event from "../../components/event/Event";

const props = {
  id: "fake-id",
  height: 100,
  marginTop: 10,
  title: "fake-title",
  time: "12:25",
  startTime: 1623187031296,
  handleDelete: jest.fn(),
  fetchEvents: jest.fn(() => Promise.resolve()),
};

describe("<Event />", () => {
  it("should render component", () => {
    const wrappedComponent = shallow(<Event {...props} />);

    expect(wrappedComponent).toMatchSnapshot();
  });

  it("should open delet modal on click", () => {
    const mockOnDelete = jest.fn();
    const wrappedComponent = shallow(<Event {...props} onClick={mockOnDelete} />);
    wrappedComponent.find(".event").simulate("click");

    expect(wrappedComponent.find(".delete-event-btn").length).toEqual(1);
  });

  it("should delet event", () => {
    const mockOnDelete = jest.fn();
    const wrappedComponent = shallow(<Event {...props} onClick={mockOnDelete} />);
    wrappedComponent.find(".event").simulate("click");
    wrappedComponent.find(".delete-event-btn").simulate("click");

    expect(props.handleDelete).toBeCalledWith("fake-id");
  });

  it("should close delete modal", () => {
    const mockOnDelete = jest.fn();
    const wrappedComponent = shallow(<Event {...props} onClick={mockOnDelete} />);
    wrappedComponent.find(".event").simulate("click").simulate("click");

    expect(props.handleDelete).toBeCalledTimes(1);
  });

  it("should not delet event", () => {
    global.alert = jest.fn();
    const mockOnDelete = jest.fn();
    props.startTime = 1629187031296;
    const wrappedComponent = shallow(<Event {...props} onClick={mockOnDelete} />);
    wrappedComponent.find(".event").simulate("click");
    wrappedComponent.find(".delete-event-btn").simulate("click");

    expect(props.handleDelete).toBeCalledWith("fake-id");
  });
});
