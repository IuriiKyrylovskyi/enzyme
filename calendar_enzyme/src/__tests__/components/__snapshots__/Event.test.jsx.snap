// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> should render component 1`] = `
<Fragment>
  <div
    className="event"
    onClick={[Function]}
    style={
      Object {
        "height": 100,
        "marginTop": 10,
      }
    }
  >
    <div
      className="event__title"
    >
      fake-title
    </div>
    <div
      className="event__time"
    >
      12:25
    </div>
  </div>
</Fragment>
`;
